runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5777,7 +5778,17 @@ class INTERPRETER_CLASS
 				omrthread_monitor_t monitor = getMonitorForWait(_currentThread, waitObject);
 				monitor->count = _currentThread->currentContinuation->waitingMonitorEnterCount;
 				_currentThread->currentContinuation->waitingMonitorEnterCount = 0;
-				returnVoidFromINL(REGISTER_ARGS, 4);
+				_currentThread->ownedMonitorCount -= 1;
+				if (J9VMJAVALANGTHREAD_DEADINTERRUPT(_currentThread, _currentThread->threadObject)) {
+					buildInternalNativeStackFrame(REGISTER_ARGS);
+					updateVMStruct(REGISTER_ARGS);
+					setCurrentException(_currentThread, J9VMCONSTANTPOOL_JAVALANGINTERRUPTEDEXCEPTION, NULL);
+					VMStructHasBeenUpdated(REGISTER_ARGS);
+					restoreInternalNativeStackFrame(REGISTER_ARGS);
```

Comment:
Shouldn't the native stack frame be left as is while throwing the exception?

Reply:
yes, will updated the code

Reply:
fixed